src/samples/PoseWithCovariance.cpp

```diff
@@ -0,0 +1,85 @@
+#include "PoseWithCovariance.hpp"
+#include <boost/format.hpp>
+#include <stdexcept>
+
+namespace base { namespace samples {
+
+PoseWithCovariance::PoseWithCovariance()
+{
+
+}
+
+PoseWithCovariance::PoseWithCovariance(const TransformWithCovariance& transform) :
+        transform(transform.translation, transform.orientation, transform.cov)
+{
+
+}
+
+PoseWithCovariance::PoseWithCovariance(const RigidBodyState& rbs) : time(rbs.time),
+    frame_id(rbs.targetFrame), object_frame_id(rbs.sourceFrame), transform(rbs.position, rbs.orientation)
+{
+    transform.cov << rbs.cov_position, Eigen::Matrix3d::Zero(),
+                        Eigen::Matrix3d::Zero(), rbs.cov_orientation;
+}
+
+PoseWithCovariance PoseWithCovariance::operator*(const PoseWithCovariance& pose) const
+{
+    // check frame ID's
+    if(this->object_frame_id.empty() || pose.frame_id.empty())
```

Comment:
Between my comments I re-read what the reason was for this operator. 
I didn't like the frame checks and the resulting necessity to throw back then and since I'm not the only one I removed the checks. Sorry @jhidalgocarrio